wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -64,6 +64,26 @@ abstract class Wallet
   val chainQueryApi: ChainQueryApi
   val creationTime: Instant = keyManager.creationTime
 
+  private def utxosWithMissingTx: Future[Vector[SpendingInfoDb]] = {
+    for {
+      utxos <- spendingInfoDAO.findAllUnspent()
+      hasTxs <- FutureUtil.foldLeftAsync(Vector.empty[SpendingInfoDb], utxos) {
+        (accum, utxo) =>
+          // If we don't have tx in our transactionDAO, add it to the list
+          transactionDAO
+            .read(utxo.txid)
+            .map(txOpt => if (txOpt.isEmpty) accum :+ utxo else accum)
+      }
+    } yield hasTxs
+  }
+
+  for {
+    utxos <- utxosWithMissingTx
+    blockHashes = utxos.flatMap(_.blockHash.map(_.flip))
+    // Download the block the tx is from so we process the block and subsequent txs
+    _ <- nodeApi.downloadBlocks(blockHashes.distinct)
+  } yield ()
```

Comment:
I think it should be in a separate initialization method (`start`, `init`, etc).

Reply:
That would be nice, however, the wallet does not have a `start` function, and `init` is only called when we are creating a new wallet.